Cinematic DOF v1.1.15.

FransBouma opened the pull request:
- Better near plane alpha calculations so pointy edges are no longer sticking out and edges of softness is now a bit translucent as it should be. 
- fixed issue with near-in focus elements in near plane which weren't blurred particularly well. 
- Changed default of focal length from 120 to 100mm

crosire commented on Shaders/CinematicDOF.fx:
```diff
@@ -442,17 +445,17 @@ namespace CinematicDOF
 		float minCoC = 10;
 		float coc;
-		float2 coordOffset = float2(BUFFER_PIXEL_SIZE.x, 0);
+		float2 coordOffset = float2(ReShade::PixelSize.x, 0);
```
It's better to use the new `BUFFER_PIXEL_SIZE` and `BUFFER_SCREEN_SIZE` defines here. This ensures better compatibility with FreeStyle (because the screen/pixel size is not a constant in FreeStyle, so the ReShade.fxh has some ugly workarounds to have `ReShade::ScreenSize`/`ReShade::PixelSize` look like global constant variables anyway, which I'd much rather get rid of in the future).

crosire replied:
Btw., you can also check whether an effect is compiled for FreeStyle by looking whether `__RESHADE_FXC__` is defined. That way you don't need to maintain two different effect variants and can instead have it all in one file.

FransBouma commented:
Regarding the repo structure, that would be possible. Not sure if you then also pull AnselSpecific as I store the ansel specific code in a separate folder. 

Regarding the constant, I'll change it. 

I use a separate shader for ansel/freestyle btw as focusing is otherwise a nightmare with the incredibly bad UI in ansel/freestyle, as the UI it generates is too imprecise (I have no idea why they do this, but if a range is wide, like 0-150, the range 0-20 is done in large steps...). The separate shader has an extra slider for range selection so you can work around the UI issues in freestyle/ansel and have proper focusing. It also has a switch which always shows the focusing overlay as it's otherwise not visible as clicks on sliders aren't seen as clicks by the shader code. 

So I have to maintain 2 separate versions anyway. That said, I have asked Henry some time ago where I had to submit a new shader for ansel's reshade support (as they have a curated set of shaders on the website) but never heard anything back.

FransBouma commented:
I've updated the shader with the desired changes (`BUFFER_PIXEL_SIZE` and `BUFFER_SCREEN_SIZE`). I've added a define for when it's not defined as well (I used reshade 4.4 in my test setup and it turns out it doesn't have the defines, so for the people who copy/paste shaders from others (or grab a setup from nexus) and use an older version, this will work too). 

I've also moved the shader/textures folders to the root of OtisFX, so it should be all set now. I haven't had time to update the shader with a define on `__RESHADE_FXC__` yet, will try to do that when I have more time. To be clear: when __RESHADE_FXC__ is defined, the shader is used in freestyle?

crosire commented:
Correct, when `__RESHADE_FXC__` is defined, it's FreeStyle. ReShade does not set that define.

crosire commented on Shaders/CinematicDOF.fx:
```diff
@@ -442,7 +452,7 @@ namespace CinematicDOF
 	float PerformTileGatherHorizontal(sampler source, float2 texcoord)
 	{
-		float tileSize = TILE_SIZE * (float(BUFFER_WIDTH) / float(GROUND_TRUTH_SCREEN_WIDTH));
+		float tileSize = TILE_SIZE * (ReShade::ScreenSize.x / GROUND_TRUTH_SCREEN_WIDTH);
```
You missed one here =)

FransBouma replied:
Hmm, I did a global search replace! :) Will correct it

FransBouma commented:
Done :)
